### ISSUE TITLE
Using useFipsEndpoint to true for Kinesis Client in Gov region builds incorrect fips-endpoint URL

### ISSUE THREAD

**Issue by sherpapasang** [aws/aws-sdk-js-v3#5907]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Kinesis DataStreams FIPS endpoint in AWS GovCloud(US)  is `kinesis.us-gov-west-1.amazonaws.com` and `kinesis.us-gov-east-1.amazonaws.com`.

Setting kinesis client in Gov regions with `useFipsEndpoint : true`, it hits a non-existing endpoint when making any calls.

### SDK version number

@aws-sdk/client-kinesis@534,

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

18.X

### Reproduction Steps

Use the aws-sdk/client-kinesis@3.534.0
configure a kinesis client in any Gov region and set `useFipsEndpoint` to true
Make any calls using the client - it will fail

### Observed Behavior

In AWS GovCloud(US)  regions, it hits a non-existing endpoint if `useFipsEndpoint` is enabled. Specifically it hits `kinesis-fips.us-gov-west-1.amazonaws.com` for `us-gov-west-1` region.

Below is the error from a lambda logs:
```
        {
            "name": "Error",
            "location": "node:internal/errors:496",
            "message": "getaddrinfo ENOTFOUND kinesis-fips.us-gov-west-1.amazonaws.com",
            "stack": "Error: getaddrinfo ENOTFOUND kinesis-fips.us-gov-west-1.amazonaws.com\n    at __node_internal_captureLargerStackTrace (node:internal/errors:496:5)\n    at __node_internal_ (node:internal/errors:715:10)\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:108:26)\n    at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:130:17)"
        }
```

### Expected Behavior

In AWS GovCloud(US)  regions, it should hit the correct fips-endpoint URL if `useFipsEndpoint` is enabled.

### Possible Solution

You can override the endpoint-url by using an environment variable `AWS_ENDPOINT_URL` or set `useFipsEndpoint` to `false`

**Comment by sherpapasang** [comment]
v2 has fixed this issue with 
https://github.com/aws/aws-sdk-js/commit/28a4e42e23850736d2aa4fc45572e6736e57e96b

**Comment by RanVaknin** [comment]
Hi @sherpapasang ,

I'm not able to reproduce this behavior. 

Here is my code:

```js
import { KinesisClient, ListStreamsCommand } from "@aws-sdk/client-kinesis";

const client = new KinesisClient({
  region: "us-gov-west-1",
  useFipsEndpoint: true
})

// middleware hook to log the outgoing request:
client.middlewareStack.add(next => args => {
    console.log('before sign: ' , args.request);
    return next(args);
}, {step: 'build'})

try {
  await client.send(new ListStreamsCommand({}))
} catch (error) {
  console.log(error)  
}
```

Will result in the request to be sent to the correct host:

```
before sign:  HttpRequest {
  method: 'POST',
  hostname: 'kinesis.us-gov-west-1.amazonaws.com',   <-- correct 
  port: undefined,
  query: {},
  headers: {
    'content-type': 'application/x-amz-json-1.1',
    'x-amz-target': 'Kinesis_20131202.ListStreams',
    'content-length': '2'
  },
  body: '{}',
  protocol: 'https:',
  path: '/',
  username: undefined,
  password: undefined,
  fragment: undefined
}
```

Can you share your complete code snippet?

Thanks,
Ran~

**Comment by sherpapasang** [comment]
Thanks for confirming @RanVaknin . I realized that I'm making the calls in NodeJS 18.X lambda. The lambda runtime includes aws-sdk v3 by default but it seems the version used is `3.188.0`. It seems the fix was released in `3.262.0` with `https://github.com/aws/aws-sdk-js-v3/commit/6af967ec2c91b641d158de174e38759a5eeec681`.

I'll close the issue. Thanks for the help.
